control-plane/agents/core/src/volume/specs.rs

```diff
@@ -472,62 +476,42 @@ impl ResourceSpecsLocked {
                 node_id: request.node.clone(),
             })?;
 
-        let specs = self.read().await;
-        if let Some(nexus_spec) = specs.get_nexus(&request.uuid) {
-            let mut spec = nexus_spec.lock().await;
-            if spec.updating {
-                return Err(SvcError::Conflict {});
-            } else if !spec.state.created() {
-                return Err(SvcError::NexusNotFound {
-                    nexus_id: request.uuid.to_string(),
-                });
-            } else if spec.share == Protocol::Off {
-                return Err(SvcError::NotShared {
-                    kind: ResourceKind::Nexus,
-                    id: request.uuid.to_string(),
-                });
-            }
+        if let Some(nexus_spec) = self.get_nexus(&request.uuid).await {
+            let spec_clone = {
+                let status = registry.get_nexus(&request.uuid).await?;
+                let mut spec = nexus_spec.lock().await;
+                if spec.pending_op() {
+                    return Err(SvcError::StoreSave {
+                        kind: ResourceKind::Nexus,
+                        id: request.uuid.to_string(),
+                    });
+                } else if spec.updating {
+                    return Err(SvcError::Conflict {});
+                } else if !spec.state.created() {
+                    return Err(SvcError::NexusNotFound {
+                        nexus_id: request.uuid.to_string(),
+                    });
+                } else if spec.share == Protocol::Off && status.share == Protocol::Off {
```

Comment:
Nit: For consistency it might be good to use the same check sequence as on line 440.

Reply:
Done